playground/miden-wasm/src/utils_debug.rs

```diff
@@ -140,57 +145,42 @@ impl DebugExecutor {
     // --------------------------------------------------------------------------------------------
 
     /// print general VM state information.
-    pub fn print_vm_state(&self) -> String {
-        format!("{}", self.vm_state)
-    }
+    fn vm_state_to_output(&self) -> DebugOutput {
+        let mut asmop = AsmOpInfo::new(String::new(), 0, 0);
+        if self.vm_state.asmop.is_some() {
+            asmop = self.vm_state.asmop.clone().unwrap();
+        }
 
-    /// print all stack items.
-    pub fn print_stack(&self) -> String {
-        let mut res =
-            self.vm_state
-                .stack
-                .iter()
-                .enumerate()
-                .fold(String::new(), |mut s, (i, f)| {
-                    s.push_str(&format!("[{i}] {f}\n"));
-                    s
-                });
-        res.pop(); // removes unnecessary line-break
-        res
+        let memory: Vec<(u64, [u64; 4])> =
+            self.vm_state.memory.iter().map(|x| (x.0, self.word_to_ints(&x.1))).collect();
+
+        let output = DebugOutput {
+            clk: self.vm_state.clk,
+            op: format!{"{:?}", self.vm_state.op},
+            asmop_op: asmop.op().to_string(),
+            asmop_num_cycles: asmop.num_cycles(),
```

Comment:
Every operation takes a single cycle, but an instruction can be composed of multiple operations. To make it clearer it is better to name this `number_of_ops` or `number_of_operations`.